.github/workflows/ci.yml

```diff
@@ -0,0 +1,59 @@
+name: Build & Tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_test:
+    # Used for installing Miri
+    # https://rust-lang.github.io/rustup-components-history/
+    env:
+      TOOLCHAIN_VERSION: '2022-09-26'
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        channel: [ "stable", "beta", "nightly" ]
+        target: [ "i686-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "aarch64-unknown-linux-gnu", "arm-unknown-linux-gnueabi", "powerpc-unknown-linux-gnu", "powerpc64-unknown-linux-gnu", "wasm32-wasi" ]
```

Comment:
Uber nit: Could you put `arm-unknown-linux-gnueabi` before `aarch64-unknown-linux-gnu` since arm 32-bit and aarch64 is 64-bit (and that's the pattern you've used for the other targets)?